source/client.c

```diff
@@ -2467,7 +2505,8 @@ static enum aws_mqtt_client_request_state s_unsubscribe_send(
         if (!timeout_task_arg) {
             return AWS_MQTT_CLIENT_REQUEST_ERROR;
         }
-        task_arg->timeout_arg = timeout_task_arg;
+        task_arg->timeout_wrapper.timeout_task_arg = timeout_task_arg;
```

Comment:
something like...

/* The request and its timeout task point to each other. Whichever one finishes first will null out both pointers. Then when the second one runs, it will know that it wasn't first */